index.bs

```diff
@@ -2387,6 +2442,84 @@ Note: Invoking this method from a [=browsing context=] where the [=Web Authentic
 
 </div>
 
+### Deserialize Registration ceremony options - PublicKeyCredential's `getCreationOptionsFromJSON()` Method ### {#sctn-getCreationOptionsFromJSON}
+
+<div link-for-hint="WebAuthentication/getCreationOptionsFromJSON">
+
+[=[WRPS]=] use this method to convert JSON-compatible serialized representations of options for
+{{CredentialsContainer/create()|navigator.credentials.create()}} to {{PublicKeyCredentialCreationOptions}}.
+
+Upon invocation, the [=client=] must take the provided options and return an identically-structured object,
+making sure to decode to {{ArrayBuffer}} those {{DOMString}} properties that are encoded with [=base64url encoding=]
+but must be {{ArrayBuffer}} values according to the definition of {{PublicKeyCredentialCreationOptions}}.
+
+{{AuthenticationExtensionsClientInputsJSON}} may include extensions registered in the IANA
+"WebAuthn Extension Identifiers" registry [[!IANA-WebAuthn-Registries]] but not defined in [[#sctn-extensions]].
+In either case the [=client=] must decode to {{ArrayBuffer}} those {{DOMString}} values that are encoded with
+[=base64url encoding=] but must be {{ArrayBuffer}} values according to the extension's definition.
```

Comment:
I think this could be a bit more rigorous, how about this?

```suggestion
[=[WRPS]=] use this method to convert [=JSON type=] representations of options for
{{CredentialsContainer/create()|navigator.credentials.create()}} into {{PublicKeyCredentialCreationOptions}}.

Upon invocation, the [=client=] MUST convert the {{PublicKeyCredential/getCreationOptionsFromJSON(options)/options}} argument
into a new, identically-structured {{PublicKeyCredentialCreationOptions}} object,
using [=base64url encoding=] to decode any {{DOMString}} attributes in {{PublicKeyCredentialCreationOptionsJSON}}
that correspond to [=buffer source type=] attributes in {{PublicKeyCredentialCreationOptions}}.
This conversion MUST extend to any [=client extension inputs=] processed by the [=client=].

{{AuthenticationExtensionsClientInputsJSON}} MAY include extensions registered in the IANA
"WebAuthn Extension Identifiers" registry [[!IANA-WebAuthn-Registries]] but not defined in [[#sctn-extensions]].
```

This honestly still feels a bit too hand-wavy to me, but it could be a step in the right direction?